ruby/example_code/s3/tests/test_auth_federation_token_request_test.rb

```diff
@@ -0,0 +1,14 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX - License - Identifier: Apache - 2.0
+
+require_relative '../auth_federation_token_request_test.rb'
+require_relative './s3_test_globals.rb'
+
+describe 'Tests whether an AWS IAM user with limited permissions can list objects in an Amazon S3 bucket' do
+  it 'Finds or creates the specified user' do
+    expect(get_user($s3_region_id, $iam_user_name)).to be
```

Comment:
we should be stubbing/mocking responses, otherwise this will make real calls and charge customers. the SDK supports stubbing (see here: https://aws.amazon.com/blogs/developer/advanced-client-stubbing-in-the-aws-sdk-for-ruby-version-3/)

```
s3 = Aws::S3::Client.new(stub_responses: true)
s3.stub_responses(:list_objects_v2, { contents: [{ key: "obj", etag: '123' }, ... ] })
```

Reply:
I changed these to use stubs/mocks, as you propose.